x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/utils.test.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import '../../../__mocks__/enterprise_search_url.mock';
+
+import { generatePreviewUrl } from './utils';
+
+jest.mock('../engine', () => ({
+  EngineLogic: {
+    values: {
+      engineName: 'national-parks-demo',
+    },
+  },
+}));
+
+describe('generatePreviewUrl', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('generates a url to the preview application from state', () => {
```

Comment:
[Optional] It may be worth documenting the new default `fromKibana` behavior in its own unit test just to highlight it for other devs.

```suggestion
  it('auto-populates the fromKibana query param', () => {
    expect(generatePreviewUrl({})).toEqual(
      'http://localhost:3002/as/engines/national-parks-demo/reference_application/preview?fromKibana=true'
    );
  });

  it('generates a url to the preview application from state', () => {
```